contrib/go-chi/chi.v5/option.go

```diff
@@ -130,3 +134,27 @@ func WithResourceNamer(fn func(r *http.Request) string) Option {
 		cfg.resourceNamer = fn
 	}
 }
+
+// WithAppsecDisabled specifies whether to enable the AppSec middleware at run-time.
+// This has not effect if AppSec is not enabled globally via the DD_APPSEC_ENABLED environment variable,
+// and is intended to allow applications to disable AppSec at runtime.
+func WithAppsecDisabled(disabled bool) Option {
+	return func(cfg *config) {
+		cfg.appsecDisabled = disabled
+	}
+}
+
+// WithAppsecOptions allows for specifying options for the AppSec middleware.
+func WithAppsecOptions(opts ...httpsec.WrapHandlerOption) Option {
+	return func(cfg *config) {
+		cfg.appsecOptions = opts
+	}
+}
+
+// WithResponseHdrFetcher provides a function to fetch the response headers from the http.ResponseWriter.
+// This allows for custom implementations as needed if you over-ride the default http.ResponseWriter.
+func WithResponseHdrFetcher(f func(http.ResponseWriter) http.Header) httpsec.WrapHandlerOption {
```

Comment:
I'm not so keen on the `Hdr` abbreviation, especially as this is a surface-level API (part of the exported methods)... I would rather this be called with the full spelling `WithResponseHeaderFetcher`).

---

In addition to this, and if I remember correctly, your custom implementation would require producing a copy of the response headers, as opposed to directly providing a reference to the underlying `http.Header`... I'm afraid this isn't clear from the naming, documentation, or type signature, which means it's not unlikely that a future use will assume they can write new headers with this... when they actually cannot.

At very minimum, I would introduce a typedef to make this clear:

```suggestion
// ResponseHeaderFetcher is a function that obtains headers from the provided http.ResponseWriter.
// Callers should assume the returned http.Header value has been deep-cloned, making
// this function only suitable for use when reading headers, not to write new ones.
type ResponseHeaderFetcher func(http.ResponseWriter) http.Reader

// WithResponseHdrFetcher provides a function to fetch the response headers from the http.ResponseWriter.
// This allows for custom implementations as needed if you over-ride the default http.ResponseWriter.
func WithResponseHeaderFetcher(f ResponseHeaderFetcher) httpsec.WrapHandlerOption {
```

Alternatively, the `Fetcher` verb could be replaced with `Copier`.